ops/main.py

```diff
@@ -334,7 +455,8 @@ def _init_dispatch(self):
         JUJU_DISPATCH_PATH will be set to the wanted hook, e.g. hooks/install,
         in both cases.
         """
-        self._dispatch_path = Path(os.environ['JUJU_DISPATCH_PATH'])
+        assert self._juju_context.dispatch_path is not None
+        self._dispatch_path = Path(self._juju_context.dispatch_path)
```

Comment:
I don't think this will be needed anymore once `_JujuContext.dispatch_path` is non-optional.